src/aiida_quantumespresso/workflows/pw/bands.py

```diff
@@ -209,14 +210,24 @@ def run_seekpath(self):
 
         This is only called if the `bands_kpoints` input was not specified.
         """
+
         inputs = {
             'reference_distance': self.inputs.get('bands_kpoints_distance', None),
             'metadata': {
                 'call_link_label': 'seekpath'
             }
         }
         result = seekpath_structure_analysis(self.ctx.current_structure, **inputs)
-        self.ctx.current_structure = result['primitive_structure']
+
+        # If the input structure was a HubbardStructureData
+        if isinstance(self.ctx.current_structure, HubbardStructureData):
+            self.ctx.hubbard_info = self.ctx.current_structure.hubbard
+            hubbard_structure = HubbardStructureData.from_structure(result['primitive_structure'])
+            hubbard_structure.hubbard = self.ctx.hubbard_info
```

Comment:
I am a bit worried here, the `habbard_info` is bind with the original structure. But the `current_structure` will then be a primitive structure. 

I think it will be more proper that there will be a `get_primitive_structure` method for `HubbardStructureData` and `hubbard_info` is always attached with the corresponded structure.

Reply:
Just passing by: be very aware that getting a primitive structure would mean to also transfer the hubbard parameters CORRECTLY, e.g. in case of extended Hubbard parameters. This is still not implemented in e.g. `HubbardUtils`, for which there is a method that is only tested for supercells. And indeed, this way would most certainly set WRONG hubbard parameters for the primitive structure, where the order of atoms might change, as well as the number, and considering different lattice parameters this is a recepy for disaster. As such, is it REALLY necessary to have a primite structure? If not, then it would be wiser to just skip this if uncertain on what it is doing.

Reply:
I wouldnt change the structure, since this function is just to pass the hubbard parameters, instead i proposed that 1D and 2D , are treated as 3D , just to pass the parameters to the HubbardStructureData,

Reply:
@unkcpz Indeed when using seekpath and generating a `primitive_structure `, we CAN'T use hubbard_info. Is also quite tricky to generalize it since the HubbardStructureData can have hubbard_parameters for DFT+U, DFT+U+V. It seems it need a method to generalize the manifold to apply it on a` primitive_structure`

Reply:
Then can you make it except when the scenario happened? If I understand correctly, with your implementation, if I run the band on the supercell with hubbard turned on, it will raise in some place that the calculation will fail. Can you try to add a validation and notify user about it.

Reply:
@unkcpz I added a logic that generalize the hubbard_info , to be transformed to the `primitive_structure`. @bastonero could you give me your opinion , I will try to make a test with a conv unit cell , asap